architecture/design/pessimistic-locking-functional-spec.md

```diff
@@ -0,0 +1,661 @@
+## [Functional Spec] YSQL Pessimistic Locking
+
+
+## 1 Introduction
+
+To understand pessimistic locking behaviour in Pg, it is important to take note of some points about row-level locking and its interaction with DMLs.
+
+<span style="text-decoration:underline;">PostgreSQL supports 4 types of row-level locks -</span>
+
+
+
+1. Exclusive modes (i.e., only 1 txn can hold such a lock at any time) -
```

Comment:
Use the full terms:
"transaction"
"primary key"
"PostgreSQL"
etc.

Reply:
done